core/modules/widgets/list.js

```diff
@@ -128,7 +129,14 @@ ListWidget.prototype.makeItemTemplate = function(title) {
 		}
 	}
 	// Return the list item
-	return {type: "listitem", itemTitle: title, variableName: this.variableName, children: templateTree};
+	var parseTreeNode = {type: "listitem", itemTitle: title, variableName: this.variableName, children: templateTree};
+	if(this.indexName) {
+		parseTreeNode.index = index.toString();
+		parseTreeNode.indexName = this.indexName;
+		parseTreeNode.isFirst = index === 0;
```

Comment:
hmm, Should we use 0-based counting for user-faced numbers? eg [nth operator](https://tiddlywiki.com/#nth%20Operator) starts with 1  ?!

Reply:
The `reduce[]` operator uses zero-based indexes, and I think it's reasonable to continue. (The `nth` operator was part of one of those PRs that I've come to regret)

Reply:
Then we should deprecate it and remove it in a year or so.